client/signup/steps/site-type/form.jsx

```diff
@@ -48,22 +48,15 @@ class SiteTypeForm extends Component {
 	handleSubmit = type => {
 		this.props.recordTracksEvent( 'calypso_signup_actions_submit_site_type', {
 			value: type,
+			user_input_site_type: this.state.otherValue || null,
 		} );
 
 		this.setState( { siteType: type } );
 
-		this.props.submitForm( type );
+		this.props.submitForm( type + '-' + this.state.otherValue );
```

Comment:
We should conditionally check for `this.state.otherValue` [like you were below](https://github.com/Automattic/wp-calypso/pull/32125/commits/cfbcc6b306642af3d4dcfadeef812888e34b88b0#diff-a15b47f96edcc7f87338c0ce465ada73L59) to prevent submitting non-"other" segments with a hyphen (i.e. `business-`).

Reply:
good point. Edited and retested.